packages/legend-query-builder/src/stores/__tests__/QueryBuilderChangeDetection.test.ts

```diff
@@ -89,5 +89,11 @@ test(integrationTest('Test change detection'), () => {
     expect(queryBuilderState.hashCode).toBe(
       queryBuilderState.changeDetectionState.hashCodeSnapshot,
     );
+
+    // Test Redo/Undo action in Query Builder
+    fireEvent.click(renderResult.getByText('Undo'));
+    await waitFor(() => getByText(projectionCols, 'Name'));
```

Comment:
Won't `getByText(projectionCols, 'Name')` still return `true` if it contains `'Legal Name'`? To check that the undo button actually changes the value from `'Legal Name'` to `'Name'`, maybe we should use a regex to ensure it only matches `'Name'`. i.e., `getByText(projectionCols, /^Name$/)`

Reply:
I think getByText looks for an exact match of the provided string within the text content of the elements.